### PR TITLE
BUG: unnecessary FutureWarning in sort_values

### LINKED ISSUE #45618: BUG: passing a SparseArray to pd.Index - overly eager FutureWarning
### Pandas version checks

- [X] I have checked that this issue has not already been reported.

- [X] I have confirmed this bug exists on the [latest version](https://pandas.pydata.org/docs/whatsnew/index.html) of pandas.

- [ ] I have confirmed this bug exists on the main branch of pandas.

### Reproducible Example

```python
In [1]: import pandas as pd
   ...: 
   ...: s = pd.Series(pd.Categorical(["a", "b", "a"], categories=["a", "b", "c"]))
   ...: exp = pd.get_dummies(s, sparse=True)
   ...: exp.sort_values(by=exp.columns.tolist())
<ipython-input-1-312167088340>:5: FutureWarning: In a future version, passing a SparseArray to pd.Index will store that array directly instead of converting to a dense numpy ndarray. To retain the old behavior, use pd.Index(arr.to_numpy()) instead
  exp.sort_values(by=exp.columns.tolist())
<ipython-input-1-312167088340>:5: FutureWarning: In a future version, passing a SparseArray to pd.Index will store that array directly instead of converting to a dense numpy ndarray. To retain the old behavior, use pd.Index(arr.to_numpy()) instead
  exp.sort_values(by=exp.columns.tolist())
<ipython-input-1-312167088340>:5: FutureWarning: In a future version, passing a SparseArray to pd.Index will store that array directly instead of converting to a dense numpy ndarray. To retain the old behavior, use pd.Index(arr.to_numpy()) instead
  exp.sort_values(by=exp.columns.tolist())
Out[1]: 
   a  b  c
1  0  1  0
0  1  0  0
2  1  0  0
```

### Issue Description

In pandas 1.4.0 there is a FutureWarning that I believe is too aggressive. In the example above it is not clear what to change in the code snippet. For context this came up in the dask test suite. 

### Expected Behavior

I imagine there might be some internal code that is coercing the array that I provided into an index. It this is the expected behavior then I think the error message needs to be changed to make it clear how to fix the code snippet.

### Installed Versions

<details>

INSTALLED VERSIONS
------------------
commit           : bb1f651536508cdfef8550f93ace7849b00046ee
python           : 3.9.9.final.0
python-bits      : 64
OS               : Linux
OS-release       : 5.11.0-7633-generic
Version          : #35~1630100930~20.04~ae2753e-Ubuntu SMP Mon Aug 30 18:23:52 UTC 
machine          : x86_64
processor        : x86_64
byteorder        : little
LC_ALL           : None
LANG             : en_US.UTF-8
LOCALE           : en_US.UTF-8

pandas           : 1.4.0
numpy            : 1.21.5
pytz             : 2021.3
dateutil         : 2.8.2
pip              : 21.3.1
setuptools       : 59.8.0
Cython           : None
pytest           : 6.2.5
hypothesis       : None
sphinx           : None
blosc            : None
feather          : None
xlsxwriter       : None
lxml.etree       : None
html5lib         : None
pymysql          : None
psycopg2         : None
jinja2           : 3.0.3
IPython          : 8.0.1
pandas_datareader: None
bs4              : None
bottleneck       : None
fastparquet      : 0.7.2
fsspec           : 2022.01.0
gcsfs            : None
matplotlib       : 3.5.1
numba            : 0.55.0
numexpr          : 2.8.0
odfpy            : None
openpyxl         : None
pandas_gbq       : None
pyarrow          : 6.0.1
pyreadstat       : None
pyxlsb           : None
s3fs             : 2022.01.0
scipy            : 1.7.3
sqlalchemy       : None
tables           : 3.6.1
tabulate         : None
xarray           : 0.20.2
xlrd             : None
xlwt             : None
zstandard        : None

</details>

### PR DESCRIPTION
**Pull request by jbrockmendel**
- [x] closes #45618
- [x] tests added / passed
- [x] Ensure all linting tests pass, see [here](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit) for how to run them
- [x] whatsnew entry

### CODE REVIEW

File: pandas/core/sorting.py [hunk 1]
```diff
@@ -320,7 +321,12 @@ def lexsort_indexer(
     keys = [ensure_key_mapped(k, key) for k in keys]
 
     for k, order in zip(keys, orders):
-        cat = Categorical(k, ordered=True)
+        with warnings.catch_warnings():
```
Comment (jreback) [inline pandas/core/sorting.py:324]: hmm does this have any perf impact?

Comment (jbrockmendel) [reply pandas/core/sorting.py:324]:
```
df = pd.DataFrame({"A": range(5), "B": range(5), "C": range(5)})

%timeit df.sort_values(["A", "B"])
486 µs ± 10.3 µs per loop (mean ± std. dev. of 7 runs, 1000 loops each)  # <- main
527 µs ± 28 µs per loop (mean ± std. dev. of 7 runs, 1000 loops each)  # <- PR
```

If we bump the length up to 10**5 its 21.7 ms vs 21.9 ms

Comment (jreback) [reply pandas/core/sorting.py:324]: ok so nothing

**Comment by jreback** [comment]
hmm failing tests though

**Comment by jbrockmendel** [comment]
> hmm failing tests though

yah places where we are testing for futurewarning and no longer getting them.  will update.

**Comment by jbrockmendel** [comment]
> yah places where we are testing for futurewarning and no longer getting them. will update.

updated + greenishx

File: doc/source/whatsnew/v1.4.1.rst [hunk 2]
```diff
@@ -28,6 +28,7 @@ Fixed regressions
 Bug fixes
 ~~~~~~~~~
 - Fixed segfault in :meth:``DataFrame.to_json`` when dumping tz-aware datetimes in Python 3.10 (:issue:`42130`)
+- Stopped emitting unnecessary ``FutureWarning`` in :meth:`DataFrame.sort_values` with sparse columns (:issue:`45618``)
```
Comment (mroeschke) [inline doc/source/whatsnew/v1.4.1.rst:31]: Nit: Extra backtick at the end of the issue number

Comment (jbrockmendel) [reply doc/source/whatsnew/v1.4.1.rst:31]: good catch, will update

**Review by mroeschke — APPROVED** [review]

**Comment by jreback** [comment]
@meeseeksdev backport 1.4.x

**Pull request merged**